core/jobs/batch_jobs/rejecting_suggestion_for_invalid_content_ids_jobs.py

```diff
@@ -0,0 +1,586 @@
+        return html
+
+    @staticmethod
+    def _update_suggestion_model(
+        suggestions: List[suggestion_models.GeneralSuggestionModel],
+        exp_model: exp_models.ExplorationModel
+    ) -> None:
+        """Updates the translation suggestion. The translation whose
+        content_id no longer exists, the suggestion status will be marked
+        as `rejected`. The RTE content of the suggestion will be updated
+        in case invalid data is present.
+
+        Args:
+            suggestions: list(GeneralSuggestionModel). A list of translation
+                suggestion models corresponding to the given exploration.
+            exp_model: ExplorationModel. The exploration model.
+
+        Returns:
+            suggestions. List[GeneralSuggestionModel]. Result containing the
+            list of updated suggestion models.
+        """
+        exp_domain = exp_fetchers.get_exploration_from_model(exp_model)
+        exp_translatable_contents = (
+            exp_domain.get_translatable_contents_collection())
+
+        translatable_content_ids = []
+        for content_id in (
+            exp_translatable_contents.content_id_to_translatable_content.keys()
+        ):
+            translatable_content_ids.append(content_id)
+
+        for suggestion in suggestions:
+            suggestion_change = suggestion.change_cmd
+            if not suggestion_change['content_id'] in translatable_content_ids:
+                suggestion.status = 'rejected'
+
+            html = suggestion_change['translation_html']
+            suggestion_change['translation_html'] = (
+                RejectSuggestionWithMissingContentIdMigrationJob._fix_content(
+                    html))
+
+        return suggestions
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns a PCollection of results from the suggestion updation.
+
+        Returns:
+            PCollection. A PCollection of results from the suggestion
+            migration.
+        """
+        target_id_to_suggestion_models = (
+            self.pipeline
+            | 'Get translation suggestion models in review' >> ndb_io.GetModels(
+                suggestion_models.GeneralSuggestionModel.get_all(
+                    include_deleted=False).filter(
+                        (
+                            suggestion_models
+                            .GeneralSuggestionModel.suggestion_type
+                        ) == feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT
+                    ).filter(
+                        suggestion_models.GeneralSuggestionModel.status == (
+                            suggestion_models.STATUS_IN_REVIEW
+                        )
+                    )
+            )
+            | 'Add target id as key' >> beam.WithKeys(  # pylint: disable=no-value-for-parameter
+                lambda model: model.target_id)
+            | 'Group exploration suggestions' >> beam.GroupByKey()
```

Comment:
Is this neede? I thought `CoGroupByKey` will group them automatically.

Reply:
Doing this was leading to the below error, so reverted back to the previous state.
```
wrapper = lambda x: [fn(x)]
  File "/home/runner/work/oppia/oppia/core/jobs/batch_jobs/rejecting_suggestion_for_invalid_content_ids_jobs.py", line 140, in <lambda>
    objects['exploration'][0]
  File "/home/runner/work/oppia/oppia/core/jobs/batch_jobs/rejecting_suggestion_for_invalid_content_ids_jobs.py", line 72, in _update_suggestion_model
    exp_domain_obj = exp_fetchers.get_exploration_from_model(exp_model)
  File "/home/runner/work/oppia/oppia/core/domain/exp_fetchers.py", line 189, in get_exploration_from_model
    'states_schema_version': exploration_model.states_schema_version,
AttributeError: 'list' object has no attribute 'states_schema_version' [while running 'Transform and migrate model']
```

Reply:
Hmm, okay no problem.